### ISSUE TITLE
New blank indented lines keeps spaces or tabs

### ISSUE THREAD

**Issue by batisteo** [atom/atom#7828]
When adding multiple indented new lines, the blank one(s) still have tabs or spaces.

```python
# Present Behaviour
class PotOfFlower(object):
····height = 12000
····
····def fall():

# Expected behaviour
class Whale(object):
····height = 12000

····def think():
```

**Comment by verma-neeraj-in** [comment]
+1

**Comment by izuzak** [comment]
@batisteo I'm not sure I understand your report, so it's hard to understand why you expected some behavior. Can you share a step-by-step explanation for reproducing the issue? For example, assume that I just started Atom and opened an untitled blank tab. What do I need to type to reproduce the problem?

**Comment by thomasjo** [comment]
@izuzak He wants the cursor to move to the right location when pressing enter after e.g. `height = 12000`, but no actual whitespace characters (spaces in this case) inserted unless he begins typing on that line.

This is "resolved" by simply saving the file though, assuming *Remove Trailing Whitespace* is enabled.

**Comment by batisteo** [comment]
Let's say the pipe is the cursor, and for the four following steps you just press enter:

```python
# Present Behaviour             # Expected behaviour
def function():|                def function():|


def function():                 def function():
····|                           ····|


def function():                 def function():
····
····|                           ····|



def function():                 def function():
····
····
····|                           ····|
```

PS: I had to disable *Remove Trailing Whitespace* to not mess my commits on huge and dirty files.

**Comment by Rob-ot** [comment]
Here's a couple gifs to explain, the orange and grey blocks you see are whitespace.

This is what Atom does, it leaves the whitespace:
![atom](https://cloud.githubusercontent.com/assets/409228/8917043/877b549c-3464-11e5-9e32-847770e7b543.gif)

This is what Sublime does, it only has the whitespace on the current line:
![sublime](https://cloud.githubusercontent.com/assets/409228/8917063/b0818460-3464-11e5-9bf6-cfd3187c9c88.gif)

This is a pretty big issue for me because manually deleting whitespace is pretty annoying. This would be fine if I had 'Remove Trailing Whitespace' turned on but it causes ugly git commits.

Thanks!

**Comment by maxbrunsfeld** [comment]
> it causes ugly git commits

Several people have this same issue with that package. Out of curiosity, don't you want trailing whitespace removed in a consistent way? It seems like the cause of the ugly git commits is the fact that the code has trailing whitespace in the first place; would the issue be solved if you just did a "Remove trailing whitespace" commit?

**Comment by Rob-ot** [comment]
Yeah, the problem is it would come back since my cowokers don't care about removing trailing whitespace nearly as much as I do. :)

**Comment by maxbrunsfeld** [comment]
Got it; that's fair. So would this be a reasonable summary of your feature request?

> When the text before my cursor on a given line contains only whitespace
> and I type a line-break
> The whitespace on that line should be deleted.

**Comment by Rob-ot** [comment]
Yeah, exactly!

Thanks Max!

**Comment by batisteo** [comment]
:+1:

**Comment by ultrasaurus** [comment]
:+1:

**Comment by jeremyramin** [comment]
I believe this is fixed by atom/whitespace#91.

**Comment by izuzak** [comment]
I think so too, @jeremyramin -- thanks for noticing.

Here's a note from that PR so that it's easier to figure out when this behavior is applied:

> For this to work, whitespace.removeTrailingWhitespace must be true and whitespace.ignoreWhitespaceOnlyLines must be false as per https://github.com/atom/atom/pull/8514#issuecomment-135127346

**Comment by batisteo** [comment]
Thanks a lot @jeremyramin !

**Comment by antcodd** [comment]
I don't believe this issue is resolved by atom/whitespace#91. An absolutely key requirement was lost: not requiring remove trailing whitespace on save to be on. It's bad enough that remove trailing whitespace is on by default. I think #8154 was a much better solution (with a different setting name) - this is an autoindent setting, nothing to do with stripping whitespace on save.

caveat: I have not tried the PR.
